### PR TITLE
Release an RxNodeComponent edge on error

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
Followup to #326 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/mrc/include/mrc/node/rx_node.hpp [hunk 1]
```diff
@@ -235,6 +235,7 @@ class RxNodeComponent : public WritableProvider<InputT>, public WritableAcceptor
                 this->get_writable_edge()->await_write(std::move(message));
             },
             [this](std::exception_ptr ptr) {
+                WritableAcceptor<OutputT>::release_edge_connection();
```
Comment (mdemoret-nv) [inline cpp/mrc/include/mrc/node/rx_node.hpp:238]: Can you add a test to `test_edges.cpp` that checks for this? Should be pretty easy to throw an error in a node component. You will get an error if its not released correctly.

**Review by mdemoret-nv — APPROVED** [review]
Approved. But I think we should add tests to check for this on node component

File: cpp/mrc/include/mrc/node/rx_node.hpp (hunk 1 above)
Comment (dagardner-nv) [reply cpp/mrc/include/mrc/node/rx_node.hpp:238]: I can, but `test_segment.cpp` or `test_node.cpp` might be better locations for this. The reason is that in order to have a context set it needs to run inside of a segment, and there are currently no tests like this in `test_edges.cpp`

Comment (mdemoret-nv) [reply cpp/mrc/include/mrc/node/rx_node.hpp:238]: lets put it in `test_node.cpp`

File: cpp/mrc/tests/test_edges.cpp [hunk 2]
```diff
@@ -278,6 +279,31 @@ class TestNodeComponent : public NodeComponent<T, T>
...
+        return base_t::make_stream([this, fn](auto&&... args) {
+            m_stream_fn_called = true;
+            return fn(std::forward<decltype(args)>(args)...);
+        });
+    }
+
+    ~TestRxNodeComponent() override
+    {
+        // Debug print
+        VLOG(10) << "Destroying TestRxNodeComponent";
```
Comment (mdemoret-nv) [inline cpp/mrc/tests/test_edges.cpp:301]: Avoid debug logs in tests

File: cpp/mrc/tests/test_edges.cpp [hunk 3]
```diff
@@ -278,6 +279,31 @@ class TestNodeComponent : public NodeComponent<T, T>
...
+    }
+
+    bool m_stream_fn_called = false;
```
Comment (mdemoret-nv) [inline cpp/mrc/tests/test_edges.cpp:304]: public members you can just drop the `m_`

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**